src/SleepRepo.js

```diff
@@ -19,43 +19,41 @@ class SleepRepo {
       }
       return arr;
     }, []);
+    return dataByUser
+  }
 
-    let avgSleepQualityPerUser = dataByUser.map(user => [...user].splice(-7 * week, 7)).map(user => user.reduce((totalQuality, day) => {
-      totalQuality += day.sleepQuality;
-      return totalQuality;
-    }, 0)).map(user => Number((user / 7).toFixed(2)));
+  returnUsersWeeklyTotals(week, metric) {
+    let dataByUser = this.organizeByUser();
+    let usersWeeklyTotals = dataByUser.map(user => [...user].splice(-7 * week, 7)).map(user => user.reduce((total, day) => {
+      total += day[metric];
+      return total;
+    }, 0))
+    return usersWeeklyTotals
+  }
 
+  returnWeeksBestSleepQuality(week, metric) {
+    let weekTotals = this.returnUsersWeeklyTotals(week, metric);
+    let weekAverages = weekTotals.map(user => Number((user / 7).toFixed(2)));
     let goodSleepers = [];
-    avgSleepQualityPerUser.forEach((user, index) => {
+    weekAverages.forEach((user, index) => {
       if (user >= 3) {
         goodSleepers.push(index + 1);
       }
     });
-    return goodSleepers;
+    return goodSleepers
+  }
+
+  returnWeeklyLongestSleepers(week, metric) {
+    let avgSleepHoursPerUser = this.returnUsersWeeklyTotals(week, metric)
+    return [Math.max(...avgSleepHoursPerUser), avgSleepHoursPerUser.indexOf(Math.max(...avgSleepHoursPerUser)) + 1];
   }
 
+
   returnLongestSleepers(date) {
     var dateData = this.sleepData.filter(day => day.date === date);
     var sortedSleepers = [...dateData].sort((a, b) => b.hoursSlept - a.hoursSlept);
     return sortedSleepers.filter(day => day.hoursSlept === sortedSleepers[0].hoursSlept).map(user => user.userID);
   }
-
-  returnWeeklyLongestSleepers(week) {
-    let dataByUser = this.sleepData.reduce((arr, user) => {
-      if (!arr[user.userID - 1]) {
-        arr[user.userID - 1] = [user];
-      } else {
-        arr[user.userID - 1].push(user);
-      }
-      return arr;
-    }, []);
-
-    let avgSleepHoursPerUser = dataByUser.map(user => [...user].splice(-7 * week, 7)).map(user => user.reduce((totalHours, day) => {
-      totalHours += day.hoursSlept;
-      return totalHours;
-    }, 0));
-    return [Math.max(...avgSleepHoursPerUser), avgSleepHoursPerUser.indexOf(Math.max(...avgSleepHoursPerUser)) + 1];
```

Comment:
This is a note for later on in our refactoring.  Line 57 is an example where they are getting the user id based on the assumption that it will always be the user objects index in the array plus one.  This work now based on our data set but in the wild where user id's might not be cronological and users may be deleted, this wouldn't work.